src/components/molecules/SearchAutocomplete/index.tsx

```diff
@@ -29,22 +61,43 @@ const SearchAutocomplete: FC<SearchAutocompleteProps> = ({ suggestions }) => {
     setSearchTerm("");
     setShowClearButton(false);
     setShowSuggestions(false);
+    if (!loading && data) {
+      setFilteredSuggestions(data.suggestions);
+    }
   };
 
-  const chooseSuggestion = (suggestion: string) => {
-    setFilteredSuggestions([]);
-    setSearchTerm(suggestion);
+  const chooseSuggestion = (suggestion: Suggestion) => {
+    if (!loading && data) {
+      setFilteredSuggestions(data.suggestions);
+    }
+    setSearchTerm(suggestion.title);
+    setTag(suggestion.title);
```

Comment:
Shouldn't we check the suggestion type before setting the tag?

Reply:
This function is used in ProblemIndex to return all problems that have this string among tags. So we don't need to know what suggestion type it is. We even don't use SUGGESTION Query here, but PROBLEMS.